React-Homeworks/src/components/Main/HeroComponent/Hero.jsx

```diff
@@ -1,84 +1,76 @@
-import React, { Component } from "react";
+import { useState, useEffect } from "react";
 import FoodCard from "../../FoodCard/FoodCard";
 import Button from "../../Buttons/Buttons";
 import { buttonsList } from "../../../__mocks__/testData";
 import "./Hero.css";
 
-class Hero extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      meals: [],
-      displayedMeals: [],
-      activeButton: 1,
-      isLoading: true,
-      itemsToShow: 6,
-    };
-  }
+const Hero = () => {
+  const [meals, setMeals] = useState([]);
+  const [displayedMeals, setDisplayedMeals] = useState([]);
+  const [activeCategory, setActiveCategory] = useState("Dessert");
+  const [isLoading, setIsLoading] = useState(true);
+  const [itemsToShow, setItemsToShow] = useState(6);
 
-  componentDidMount() {
-    fetch("https://65de35f3dccfcd562f5691bb.mockapi.io/api/v1/meals")
+  useEffect(() => {
+    setIsLoading(true);
+    setDisplayedMeals([]); // Очистка отображаемых блюд при начале загрузки
+    fetch(
+      `https://65de35f3dccfcd562f5691bb.mockapi.io/api/v1/meals?category=${activeCategory}`,
+    )
       .then((response) => response.json())
       .then((data) => {
-        this.setState({
-          meals: data,
-          isLoading: false,
-          displayedMeals: data.slice(0, this.state.itemsToShow),
-        });
+        setMeals(data);
+        setTimeout(() => {
+          setDisplayedMeals(data.slice(0, itemsToShow));
+          setIsLoading(false);
+        }, 300); // Минимальная задержка для демонстрации анимации
       })
-      .catch((error) => console.error("Error fetching meals:", error));
-  }
+      .catch((error) => {
+        console.error("Error fetching meals:", error);
+        setIsLoading(false);
+      });
+  }, [activeCategory]);
```

Comment:
You don't need to fetch the meal list every time the filter is changed. You should upload them once, store it and rest of manipulations make with the stored list